src/config/voice.c

```diff
@@ -39,44 +40,43 @@ static const char* getfield(char* value, int num) {
 
 static int ini_handler(void* user, const char* section, const char* name, const char* value)
 {
-    (void) user;
+    u16 *req_id = user;
     char tmp[100];
     strlcpy(tmp, value, 100);
     int duration = atoi(getfield(tmp,2));
-#if HAS_MUSIC_CONFIG
-    char label[MAX_VOICE_LABEL];
-    strlcpy(label,getfield(tmp, 1), MAX_VOICE_LABEL);
-#endif
+    u16 id = atoi(name);
 
-    if (MATCH_SECTION(SECTION_VOICE_GLOBAL)) {
-        for (int i = 0; i < CUSTOM_ALARM_ID; i++) {
-            snprintf(tempstring, 4, "%d", i);
-            if (MATCH_KEY(tempstring)) {
-                voice_map[i].duration = duration;
-                voice_map[i].id = i;
 #if HAS_MUSIC_CONFIG
-                strcpy(voice_map[i].label, label);
+    if ( (*req_id != MAX_VOICEMAP_ENTRIES) && (*req_id == id) ) {
+        strlcpy(tempstring,getfield(tmp, 1), MAX_VOICE_LABEL);
+        return 1;
+    }
 #endif
-                return 1;
+    if ( *req_id == MAX_VOICEMAP_ENTRIES ) {
+        if (MATCH_SECTION(SECTION_VOICE_GLOBAL)) {
+            for (int i = 0; i < CUSTOM_ALARM_ID; i++) {
+                snprintf(tempstring, 4, "%d", i);
+                if (MATCH_KEY(tempstring)) {
+                    voice_map[i].duration = duration;
+                    voice_map[i].id = i;
+                    return 1;
+                }
             }
         }
+        if (MATCH_SECTION(SECTION_VOICE_CUSTOM)) {
+            voice_map[voice_map_entries].duration = duration;
+            voice_map[voice_map_entries].id = atoi(name);
+            voice_map_entries++;
+            return 1;
+        }
+        printf("Unknown entry in voice.ini: %s\n", value);
+        return 0;
     }
-    if (MATCH_SECTION(SECTION_VOICE_CUSTOM)) {
-        voice_map[voice_map_entries].duration = duration;
-        voice_map[voice_map_entries].id = atoi(name);
-#if HAS_MUSIC_CONFIG
-        strcpy(voice_map[voice_map_entries].label, label);
-#endif
-        voice_map_entries++;
-        return 1;
-    }
-    printf("Unknown entry in voice.ini: %s\n", value);
-    return 0;
+    return 1; // voice label ignored
 }
 
-void CONFIG_VoiceParse()
+void CONFIG_VoiceParse(u16 id)
 {
```

Comment:
why use a u16 here?  It is almost always beter to use 'unsigned' for passed variables (and only use the reduced size variables for global/static